### PR TITLE
feat(ci): add markdown link validation to CI

### PR DESCRIPTION
**Pull request by galligan**
## Summary

- Registers `check-markdown-links` as a subcommand in the `@outfitter/tooling` CLI (for external consumers via `bunx @outfitter/tooling check-markdown-links`)
- Adds `markdown-links` subcommand to `outfitter repo check` (for monorepo invocation via source-first resolution)
- Wires `docs:check:links` in root `package.json` through `outfitter repo check markdown-links` to comply with boundary invocation rules
- Adds export for `./cli/check-markdown-links` in `@outfitter/tooling` package.json
- Full test coverage for the link validator

## Test plan

- [ ] `bun run docs:check:links` validates relative links in docs
- [ ] `bun run apps/outfitter/src/commands/repo.ts check markdown-links --cwd .` works
- [ ] `bun run check-boundary-invocations` passes (no direct `packages/*/src` invocations)
- [ ] External consumers can use `bunx @outfitter/tooling check-markdown-links`

Closes OS-286

Closes: OS-286

<h3>Greptile Summary</h3>

This PR integrates markdown link validation into the Outfitter CI pipeline. The implementation follows the project's architecture by exposing the check-markdown-links tool through multiple surfaces: as a standalone CLI command in `@outfitter/tooling`, as a subcommand in `outfitter repo check`, and as a root package script.

Key changes:
- Pure function design with `extractMarkdownLinks()` and `validateLinks()` enables comprehensive unit testing
- Robust inline code span handling using a proper backtick-counting parser that correctly handles single, double, and triple backtick code spans
- Code fence detection to skip links in fenced code blocks
- Automatic exclusion of `docs/packages/` paths (gitignored generated content)
- Full test coverage with 16 tests covering extraction logic, filesystem validation, and edge cases
- CI integration via `docs:check:links` script in the verification pipeline
- Package boundary compliance by routing through `outfitter repo check markdown-links`

The implementation addresses the backtick handling issues mentioned in previous review threads by replacing the simple regex with a character-by-character parser that properly matches opening and closing backtick runs.

<h3>Confidence Score: 5/5</h3>

- This PR is safe to merge with no identified issues
- Clean implementation with full test coverage (16 tests), proper architectural integration following project patterns, and resolution of previously identified inline code handling issues. All boundary invocation rules are followed, and the change is additive with no risk to existing functionality.
- No files require special attention

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/tooling/src/cli/check-markdown-links.ts | New markdown link validator with comprehensive inline code handling and test coverage |
| packages/tooling/src/__tests__/check-markdown-links.test.ts | Comprehensive test suite covering extraction, validation, and edge cases |
| packages/tooling/src/cli/index.ts | Registers check-markdown-links command in tooling CLI |
| apps/outfitter/src/commands/repo.ts | Adds markdown-links subcommand to repo check group |
| packages/tooling/package.json | Exports check-markdown-links CLI entrypoint for external consumers |
| package.json | Adds docs:check:links script to CI verification pipeline |
| packages/presets/README.md | New README documenting presets package purpose and usage |

</details>

</details>

<sub>Last reviewed commit: 48ccf2c</sub>

### CODE REVIEW

File: packages/tooling/src/cli/check-markdown-links.ts [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+				!(fenceMatch[4] ?? "").trim()
+			) {
+				inCodeFence = false;
+				continue;
+			}
+		}
+		if (inCodeFence) continue;
+
+		// Strip inline code before extracting links
+		const stripped = line.replace(/`[^`]+`/g, "");
```
[the remark this answers is not in the document]
Comment (galligan) [reply packages/tooling/src/cli/check-markdown-links.ts]: Addressed in subsequent commits on this stack; this thread is now outdated relative to the current diff. Resolving.
